compiler/rustc_incremental/src/persist/file_format.rs

```diff
@@ -35,6 +36,8 @@ pub(crate) fn write_file_header(stream: &mut FileEncoder, sess: &Session) {
     assert_eq!(rustc_version.len(), (rustc_version.len() as u8) as usize);
     stream.emit_raw_bytes(&[rustc_version.len() as u8]);
     stream.emit_raw_bytes(rustc_version.as_bytes());
+    // Reserve space for the u64 where we will later write in the expected total size of this file.
+    stream.emit_raw_bytes(&0u64.to_le_bytes());
```

Comment:
Would it be simpler to just append some magic bytes like `b"the-end-of-rust"` or something like that?` Then all the `metadata` calls wouldn't be needed.

Reply:
I'd rather report the expected and actual numbers so that we can tell if the dep graph is truncated, unexpectedly large, and about how big it is. Some systems have file size limits that it's plausible we run into, and having a number would let us know about that.